documentation/src/main/asciidoc/userguide/chapters/query/hql/QueryLanguage.adoc

```diff
@@ -1022,13 +1025,33 @@ Next, functions for working with numeric values:
 | `exp()` | Exponential function | `exp(x)` | Universal in SQL dialects
 | `power()` | Exponentiation | `power(x,y)` | Universal in SQL dialects
 | `ln()` | Natural logarithm | `ln(x)` | Very common in SQL dialects
+| `log()` | Base-N logarithm | `log(x,x)` | Very common in SQL dialects
+| `log2()` | Base-2 logarithm | `log2(x)` | Very common in SQL dialects
 | `log10()` | Base-10 logarithm | `log10(x)` | Very common in SQL dialects
-| `sin()`, `cos()`, `tan()`, `asin()`, `acos()`, `atan()`, `atan2()` | Basic trigonometric functions | `sin(theta)`, `cos(theta)`, `atan2(opposite, adjacent)` | Very common in SQL dialects
+| `sin()`, `cos()`, `tan()`, `cot()`, `asin()`, `acos()`, `atan()`, `atan2()`, `acot()` | Basic trigonometric functions | `sin(theta)`, `cos(theta)`, `atan2(opposite, adjacent)` | Very common in SQL dialects
+| `sinh()`, `cosh()`, `tanh()`, `coth()`, `asinh()`, `acosh()`, `atanh()`, `acoth()` | Basic trigonometric functions | `sinh(theta)`, `cosh(theta)` | Very common in SQL dialects
 | `round()` | Numeric rounding | As usual: `round(number, places)` | Very common in SQL dialects
 | `floor()` | Floor function | `floor(x)` | Universal in SQL dialects
 | `ceiling()` | Ceiling function | `ceiling(x)` | Very common in SQL dialects
 | `least()` | Return the smallest of the given arguments | `least(x, y, z)` | Very common in SQL dialects
 | `greatest()` | Return the largest of the given arguments | `greatest(x, y, z)` | Very common in SQL dialects
+| `degrees()` | Converts radians to degrees | `degrees(x)` | Very common in SQL dialects
+| `radians()` | Converts degrees to radians | `radians(x)` | Very common in SQL dialects
+| `pi()` | The pi constant | `pi()` | Very common in SQL dialects
+| `rand()` | Return random double between 0 and 1 | `rand()` | Very common in SQL dialects
+|===
+
+Also, some bit operation functions:
+
+|===
+| HQL Function | Purpose | Syntax | Notes on syntax
+
+| `bitand()` | Bitwise and i.e. `&` | `bitand(x,y)` | Standard ANSI SQL
+| `bitor()` | Bitwise or i.e. `\|` | `bitor(x,y)` | Standard ANSI SQL
+| `bitxor` | Bitwise xor i.e. `^` | `bitxor(x,y)` | Standard ANSI SQL
+| `bitnot()` | Bitwise not i.e. `~` | `bitnot(x)` | Standard ANSI SQL
+| `octet_length()` | The length in octets/bytes | `octet_length(x)` | Standard ANSI SQL
+| `bit_length()` | The length in bits | `bit_length(x)` | Standard ANSI SQL
```

Comment:
Rollback this change too, please.

Reply:
1. I have no idea if `bitxor()` and `bitnot()` are implemented in a portable way, but for sure we have no tests for them. Are these *really* part of standard ANSI SQL??
2. I don't see why HQL needs `octet_length()` and `bit_length()` functions, and I can't imagine what you would use them for. There do seem to be a couple of very minimal tests for these, so maybe they are already portable. OTOH, they have ugly `_`s in their names, unlike our other functions. Ultimately, I'm against adding them to the list.